### PR TITLE
MUI Component Added: Universal Button

### PR DESCRIPTION
**Pull request by evanjscallan**
Reusable button component added using Material-UI. 
**Details:**
-May need color alterations to fit style sheet. 
-Text data is fed from _app.js to customize button text via array. 
-Currently implemented into index.js and will be migrated over to the other pages in future fixes.

### CODE REVIEW

**Review by prafulfillment — APPROVED** [review]

File: pages/_app.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import '../styles/globals.css'
 
 import Layout from '../components/Layout'
+const elements = ['Learn More', 'Get Involved']
```
Comment (paulcretu) [inline pages/_app.js:4]: This probably doesn't belong here, the `_app.js` file is for setting up the layout of the Next app. Anything on this page will stay loaded between page changes, which is the main purpose of including things in here. I wouldn't export anything other the the default export. What are you trying to achieve? Can try to help you find somewhere better for this variable. It could probably go right in `index.js` where you're using it.

File: components/MainButton.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      dark: '#ba000d',
+      contrastText: '#000',
+    },
+  },
+})
+
+//receives 'buttonElements' from index.js -- Custom text can be passed using the elements array. //
+const MainButton = ({ elements }) => {
+  return (
+    <div>
```
Comment (paulcretu) [inline components/MainButton.js:27]: No need for the extra `div`

File: components/MainButton.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      <Button
+        variant="contained"
+        sx={{
+          margin: '1rem',
+          bgcolor: '#101738 !important',
```
Comment (paulcretu) [inline components/MainButton.js:32]: Generally it's good to avoid using `!important` because it's really hard to override. Is it truly necessary here? Also consider moving the color into the theme, setting one-off colors leads to inconsistency

File: components/MainButton.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    secondary: {
+      light: '#ff7961',
+      main: 'purple',
+      dark: '#ba000d',
+      contrastText: '#000',
+    },
+  },
+})
+
+//receives 'buttonElements' from index.js -- Custom text can be passed using the elements array. //
```
Comment (paulcretu) [inline components/MainButton.js:24]: Awesome job leaving a comment. Just a minor note, if this button is re-used, the text might come from somewhere else than just `index.js`

File: pages/learn.js [hunk 5]
```diff
@@ -12,6 +12,8 @@ import salishsea from '../images/salishsea.png'
 import audio from '../public/frequency.mp3'
 import orcasound from '../public/frequency.mp3'
 import learnStyles from '../styles/Learn.module.css'
+
+
```
Comment (paulcretu) [inline pages/learn.js:16]: Unnecessary whitespace changes (this file shouldn't be in the PR since there are no other edits to it)

**Review by paulcretu — CHANGES_REQUESTED** [review]
Thanks for your first PR @evanjscallan! And for your patience, sorry it took me so long to review this. The code looks pretty good overall. I will say that if you just want to set up a default style for a button, there are a few other approaches that may be simpler than making a separate component (for example, using [MUI `styled()`](https://mui.com/customization/how-to-customize/#2-reusable-style-overrides)) or doing a [global override](https://mui.com/customization/theme-components/#overrides-based-on-props) to set up a "variant" of a component.

Regardless, I reviewed the code and left a ton of comments, hope it's useful feedback. Most of them are minor, the only larger issues are exporting from `_app.js` and some changes that could be omitted from this PR.

I just merged a lot of other changes so now there are merge conflicts. So if you want to get this PR merged you'll have to update it to match the latest changes in the base branch (sorry about that, lmk if you need me to help out since I kinda broke it).

File: components/MainButton.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import styled from '@material-ui/core/styles'
+import PropTypes from 'prop-types'
+import Button from '@material-ui/core/Button'
+import { createTheme } from '@material-ui/core/styles'
+
+//custom theme for coloring
+const theme = createTheme({
```
[the remark this answers is not in the document]
Comment (evanjscallan) [reply components/MainButton.js:7]: Switching to { useTheme } instead.

File: components/MainButton.js (hunk 4 above)
Comment (evanjscallan) [reply components/MainButton.js:24]: Going to eliminate the array from now and just pass string props to the buttons to simplify.

File: components/MainButton.js (hunk 3 above)
Comment (evanjscallan) [reply components/MainButton.js:32]: Removed, utilizing theme.ts now. Also changed the naming from 'MainButton' to ActionButton.

File: next-env.d.ts [hunk 7]
```diff
@@ -1,3 +1,6 @@
 /// <reference types="next" />
 /// <reference types="next/types/global" />
 /// <reference types="next/image-types/global" />
+
+// NOTE: This file should not be edited
```
[the remark this answers is not in the document]
Comment (evanjscallan) [reply next-env.d.ts:5]: Did a fresh clone/pull as I had some breaking changes from a while back. May be creating a new pull request.

File: package.json [hunk 8]
```diff
@@ -15,7 +15,8 @@
     "@emotion/styled": "^11.3.0",
     "@material-ui/core": "^5.0.0-beta.0",
     "bootstrap": "^4.6.0",
-    "next": "11.0.1",
+    "next": "^11.1.3",
```
[the remark this answers is not in the document]
Comment (evanjscallan) [reply package.json:18]: See previous comment. Fresh clone

File: package.json [hunk 9]
```diff
@@ -15,7 +15,8 @@
...
+    "prop-types": "^15.7.2",
```
[the remark this answers is not in the document]
Comment (evanjscallan) [reply package.json:19]: Removed

File: pages/_app.js (hunk 1 above)
Comment (evanjscallan) [reply pages/_app.js:4]: Removed, using string props instead to simplify.

File: pages/learn.js (hunk 5 above)
Comment (evanjscallan) [reply pages/learn.js:16]: Reverted

**Comment by paulcretu** [comment]
Closing, superseded by #55
